SimTracker/SiPhase2Digitizer/plugins/PixelDigitizerAlgorithm.cc

```diff
@@ -203,3 +211,62 @@ void PixelDigitizerAlgorithm::add_cross_talk(const Phase2TrackerGeomDetUnit* pix
     }
   }
 }
+
+PixelDigitizerAlgorithm::TimewalkCurve::TimewalkCurve(const edm::ParameterSet& pset)
+    : x_(pset.getParameter<std::vector<double>>("charge")), y_(pset.getParameter<std::vector<double>>("delay")) {
+  if (x_.size() != y_.size())
+    throw cms::Exception("Configuration") << "Timewalk model error: Invalid curve.";
```

Comment:
could make this message a bit more informative, like the one below: "the number of charge values does not match the number of delay values"